src/components/TaskStatusBadge/index.tsx

```diff
@@ -70,19 +53,54 @@ const TaskStatusBadge: React.FC<TaskStatusBadgeProps> = ({ status }) => {
     );
   }
 
-  if (status in mapBadgeColors) {
-    return (
-      <StyledBadge
-        data-cy="task-status-badge"
-        key={status}
-        {...mapBadgeColors[status]}
-      >
-        {displayStatus}
-      </StyledBadge>
-    );
-  }
+  return (
+    <StyledBadge
+      data-cy="task-status-badge"
+      key={status}
+      {...customBadgeColors(status)}
+    >
+      {displayStatus}
+    </StyledBadge>
+  );
+};
+
+const mapTaskStatusToBadgeVariant = {
+  [TaskStatus.Inactive]: Variant.LightGray,
+  [TaskStatus.Unstarted]: Variant.LightGray,
+  [TaskStatus.Undispatched]: Variant.LightGray,
+  [TaskStatus.Blocked]: Variant.LightGray,
+  [TaskStatus.Pending]: Variant.LightGray,
+  [TaskStatus.Unscheduled]: Variant.LightGray,
+  [TaskStatus.Aborted]: Variant.LightGray,
+  [TaskStatus.Started]: Variant.Yellow,
+  [TaskStatus.Dispatched]: Variant.Yellow,
+  [TaskStatus.Failed]: Variant.Red,
+  [TaskStatus.TestTimedOut]: Variant.Red,
+  [TaskStatus.TaskTimedOut]: Variant.Red,
+  [TaskStatus.Succeeded]: Variant.Green,
+  [TaskStatus.KnownIssue]: Variant.Red,
+  [TaskStatus.WillRun]: Variant.DarkGray,
+};
+const customBadgeColors = (status: string) => {
```

Comment:
This would require updating a few components, but I think it would be ideal to have `status` typed as a `TaskStatus`. I believe if you change [this line](https://github.com/evergreen-ci/spruce/blob/main/src/pages/Task.tsx#L104) to read

```js
<TaskStatusBadge status={TaskStatus[status]} />
```

and update its child components, we should be able to use `TaskStatus` safely.

Reply:
`TaskStatus[status]` doesn't actually work. 
`status` is the value returned from the server which represents a value in the `TaskStatus` enum so we can't actually use at as a key for it. 

```
TaskStatus["success"] === undefined 
TaskStatus.Success === "success"
```